_posts/blog/2016-07-07-reactive_mindset_burgers.md

```diff
@@ -0,0 +1,165 @@
+author: florina_muntenescu
+excerpt: What do Donald Duck and reactive programming have in common? Burgers! Check out why.
+tags: [Android, RxJava]
+date: 2016-07-16T15:39:55-04:00
+---
+Here's another way of explaining reactive programming: this time with Donald Duck and his nephews.
+
+## Donald Duck and His Little Helpers
+
+Donald Duck decided to switch from cinema to the food industry and open up a burger joint. But, he's not doing this alone; his three nephews - Huey, Dewey and Louie - are helping him. Each of them is doing exactly one thing:
+
+* Huey gives Donald Duck buns.
+* Dewey gives Donald Duck pieces of raw meat
+* Louie gives Donald slices of tomatoes
+
+<center>
+<picture>
+	<a href="/images/blog/reactive_mindset_burgers/nephews.png"><img src="/images/blog/reactive_mindset_burgers/nephews.png" alt="Huey, Dewey and Louie" width="350"></a>
+	<figcaption>Huey, Dewey and Louie</figcaption>
+</picture>
+</center>
+
+As soon as the restaurant opens, the nephews start working, continuously supplying Donald with buns, meat and tomato slices.
+
+With every new piece of meat that Donald gets, he first needs to make sure that the meat hasn't gone bad, and use only the fresh meat for his burgers. Then, every piece of fresh meat needs to be cooked.
+
+<center>
+<picture>
+	<a href="/images/blog/reactive_mindset_burgers/filter_map_meat.png"><img src="/images/blog/reactive_mindset_burgers/filter_map_meat.png" alt="filter and map" width="350"></a>
+	<figcaption>Filter the meat pieces and cook them</figcaption>
+</picture>
+</center>
+
+The next step is to make the burger. Donald needs to take the first bun given by Huey, together with the first piece of cooked meat and with the first slice of tomato given by Louie and then "zip" together the first burger.
+
+For the second burger, he will use the second bun from Huey, the second piece of cooked meat and the second slice of tomato.The time took by each of the nephews to do his part is different. Louie will be providing tomato slices faster than Dewey can cook the meat. As a result, Donald has to wait for it to be ready, in order to make the burger. Because he doesn't want to throw away any tomato slices, he will always use the first one on the row, given by Louie.
+
+<center>
+<picture>
+	<a href="/images/blog/reactive_mindset_burgers/zip_burgers.png"><img src="/images/blog/reactive_mindset_burgers/zip_burgers.png" alt="zip burgers" width="350"></a>
+	<figcaption>Combine the bun, the meat and the slice of tomato to make the burger</figcaption>
+</picture>
+</center>
+
+## ReactiveX and Buns, Meat & Tomato Slices
+
+In ReactiveX the buns are streams of data. The meat and the tomato slices are also streams of data, and the burgers also, but created by composing multiple streams of data. The class that emits a stream of data is called ``Observable``. So this means that we will have an Observable of buns, an Observable of meat, an Observable of tomato slices and in the end, an Observable of burgers.
```

Comment:
In ReactiveX the buns, tomato slices and the meat are streams of data. By composing multiple streams  we create a new one, composed of burgers.